service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GetGpcDocumentReferencesTaskExecutor.java

```diff
@@ -0,0 +1,123 @@
+import uk.nhs.adaptors.gp2gp.common.task.TaskDispatcher;
+import uk.nhs.adaptors.gp2gp.common.task.TaskExecutor;
+import uk.nhs.adaptors.gp2gp.ehr.EhrExtractStatusService;
+import uk.nhs.adaptors.gp2gp.gpc.builder.GpcRequestBuilder;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hl7.fhir.dstu3.model.Attachment;
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.DocumentReference;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class GetGpcDocumentReferencesTaskExecutor implements TaskExecutor<GetGpcDocumentReferencesTaskDefinition> {
+
+    @Autowired
+    private EhrExtractStatusService ehrExtractStatusService;
+    @Autowired
+    private GpcRequestBuilder gpcRequestBuilder;
+    @Autowired
+    private GpcClient gpcClient;
+    @Autowired
+    private TaskDispatcher taskDispatcher;
+
+    @Override
+    public Class<GetGpcDocumentReferencesTaskDefinition> getTaskType() {
+        return GetGpcDocumentReferencesTaskDefinition.class;
+    }
+
+    @Override
+    public void execute(GetGpcDocumentReferencesTaskDefinition taskDefinition) {
+        LOGGER.info("Execute called from GpcFindDocumentsTaskExecutor");
+
+        String patientId = retrievePatientId(taskDefinition);
+        ehrExtractStatusService.updateEhrExtractStatusAccessDocumentPatientId(taskDefinition, patientId);
+
+        if (!patientId.isBlank()) {
+            List<String> urls = retrieveDocumentReferences(taskDefinition, patientId);
+            ehrExtractStatusService.updateEhrExtractStatusAccessDocumentDocumentReferences(taskDefinition, urls);
+
+            urls.forEach(url -> queueGetDocumentsTask(taskDefinition, url));
+        }
+
+    }
+
+    private String retrievePatientId(GetGpcDocumentReferencesTaskDefinition taskDefinition) {
+        var request = gpcRequestBuilder.buildGetPatientIdentifierRequest(taskDefinition);
+        var response = gpcClient.getPatientRecord(request, taskDefinition);
+
+        FhirContext ctx = FhirContext.forDstu3();
+        IParser parser = ctx.newJsonParser();
+
+        if (response.contains("Bundle")) {
+            Bundle fhirBundle = parser.parseResource(Bundle.class, response);
+            if (!fhirBundle.getEntry().isEmpty()) {
+                Patient patient = (Patient) fhirBundle.getEntry().get(0).getResource();
+                return patient.getIdElement().getIdPart();
+            }
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private List<String> retrieveDocumentReferences(GetGpcDocumentReferencesTaskDefinition taskDefinition, String patientId) {
+        var request = gpcRequestBuilder.buildGetPatientDocumentReferences(taskDefinition, patientId);
+        var response = gpcClient.getDocumentReferences(request, taskDefinition);
+
+        if (response.contains("Bundle")) {
```

Comment:
Do we expect other responses than ones with a  "Bundle" resource?

Reply:
Nope, I'd remove this check and just parse it assuming its a Bundle resource.

Reply:
removed